### PR TITLE
Implement cascading delete

### PR DESCRIPTION
**Pull request by jirevwe**

### CODE REVIEW

File: datastore/mongo/mongo_test.go [hunk 1]
```diff
@@ -15,7 +16,9 @@ import (
 )
 
 func getDSN() string {
-	return os.Getenv("TEST_MONGO_DSN")
+	e := os.Getenv("TEST_MONGO_DSN")
+	fmt.Println(e)
```
Comment (Dotunj) [inline datastore/mongo/mongo_test.go:20]: 👀

File: datastore/mongo/group.go [hunk 2]
```diff
@@ -173,13 +180,44 @@ func (db *groupRepo) FillGroupsStatistics(ctx context.Context, groups []*datasto
...
+			return err
+		}
+
+		for _, app := range apps {
+			err = db.deleteAppEvents(sessCtx, uid, updateAsDeleted)
+			if err != nil {
+				return err
+			}
+
+			err = db.deleteApp(sessCtx, app.UID, updateAsDeleted)
```
Comment (Dotunj) [inline datastore/mongo/group.go:211]: Shouldn't we delete the app's subscription as well?

File: datastore/mongo/mongo_test.go (hunk 1 above)
Comment (jirevwe) [reply datastore/mongo/mongo_test.go:20]: Yeah, I'm trying to understand why the tests are failing on CI but passing locally

File: datastore/mongo/group.go (hunk 2 above)
Comment (jirevwe) [reply datastore/mongo/group.go:211]: Yeah, it's becoming a mess though, lol, I'll add it

**Review by Dotunj — APPROVED** [review]

**Review by danvixent — APPROVED** [review]

**Pull request merged**